src/Text/Pandoc/Readers/LaTeX.hs

```diff
@@ -547,8 +547,17 @@ coloredInline stylename = do
   color <- braced
   spanWith ("",[],[("style",stylename <> ": " <> untokenize color)]) <$> tok
 
-ttfamily :: PandocMonad m => LP m Inlines
-ttfamily = code . stringify . toList <$> tok
+ttfamily :: Inlines -> Inlines
+ttfamily =
+  fromList
+    . concatMap (toList . fmt)
+    . groupBy (\a b -> not (isStrong a || isStrong b))
+    . toList
+  where
+    isStrong (Strong _) = True
+    isStrong _ = False
+    fmt [Strong x] = strong $ code $ stringify x
+    fmt xs = code $ stringify xs
```

Comment:
This will certainly work for the special case of textbf inside texttt.  But I wonder if we couldn't generalize it a bit?  After all, someone might have emph or textsc or any number of other kinds of inline formatting inside texttt...

Reply:
One option would be to group Str and Space and SoftBreak (maybe other) elements together into chunks, and leave other inlines separate.  Then:

- for the chunks consisting entirely of Str, Space, SoftBreak, you simply put them inside Code.
- for the other chunks, use `walk` to do the same transformation recursively on their `[Inline]` children.

Reply:
Indeed, this code seems general enough to be used in other readers, and maybe could go in Shared?  I have a vague memory that we do something like this already in one of the other readers, but I can't recall which.